From 4489619110dc0d265cd03c45f6a0143b085ff772 Mon Sep 17 00:00:00 2001
From: Poonam Aggrwal <poonam.aggrwal@freescale.com>
Date: Thu, 20 Sep 2012 19:41:02 +0000
Subject: [PATCH 088/227] B4860: Added FMAN nodes in B4860 device tree

Only 1G nodes added all SGMII. SGMII1 to SGMII6.
PHY links are set as "fixed-link" for now.

Signed-off-by: Vakul Garg <vakul@freescale.com>
Signed-off-by: Poonam Aggrwal <poonam.aggrwal@freescale.com>
Signed-off-by: Sandeep Singh <Sandeep@freescale.com>
Signed-off-by: Andy Fleming <afleming@freescale.com>
[Kevin: Original patch taken from fsl sdk 1.3.1
QorIQ-SDK-V1.3.1-SOURCE-20121220-yocto.iso.]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/boot/dts/b4860qds.dts          |   63 ++++++++++++++++++++++++++-
 arch/powerpc/boot/dts/fsl/b4860si-post.dtsi |   35 +++++++++++++++
 arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi  |   12 +++++
 3 files changed, 109 insertions(+), 1 deletion(-)

diff --git a/arch/powerpc/boot/dts/b4860qds.dts b/arch/powerpc/boot/dts/b4860qds.dts
index 4ffab99..e10f8b2 100644
--- a/arch/powerpc/boot/dts/b4860qds.dts
+++ b/arch/powerpc/boot/dts/b4860qds.dts
@@ -41,6 +41,14 @@
 	#size-cells = <2>;
 	interrupt-parent = <&mpic>;
 
+	aliases{
+		ethernet0 = &enet0;
+		ethernet1 = &enet1;
+		ethernet2 = &enet2;
+		ethernet3 = &enet3;
+		ethernet4 = &enet4;
+		ethernet5 = &enet5;
+	};
 
 	ifc: localbus@ffe124000 {
 		reg = <0xf 0xfe124000 0 0x2000>;
@@ -153,6 +161,33 @@
 			dr_mode = "host";
 			phy_type = "ulpi";
 		};
+
+		fman0: fman@400000 {
+			enet0: ethernet@e0000 {
+				fixed-link = <1 1 1000 0 0 >;
+				phy-connection-type = "sgmii";
+			};
+			enet1: ethernet@e2000 {
+				fixed-link = <2 1 1000 0 0 >;
+				phy-connection-type = "sgmii";
+			};
+			enet2: ethernet@e4000 {
+				fixed-link = <3 1 1000 0 0 >;
+				phy-connection-type = "sgmii";
+			};
+			enet3: ethernet@e6000 {
+				fixed-link = <4 1 1000 0 0 >;
+				phy-connection-type = "sgmii";
+			};
+			enet4: ethernet@e8000 {
+				fixed-link = <5 1 1000 0 0 >;
+				phy-connection-type = "sgmii";
+			};
+			enet5: ethernet@ea000 {
+				fixed-link = <6 1 1000 0 0 >;
+				phy-connection-type = "sgmii";
+			};
+		};
 	};
 
 	pci0: pcie@ffe200000 {
@@ -169,7 +204,33 @@
 				  0 0x00010000>;
 		};
 	};
-
+	fsl,dpaa {
+		compatible = "fsl,b4860-dpaa", "fsl,dpaa";
+		ethernet@0 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&enet0>;
+		};
+		ethernet@1 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&enet1>;
+		};
+		ethernet@2 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&enet2>;
+		};
+		ethernet@3 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&enet3>;
+		};
+		ethernet@4 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&enet4>;
+		};
+		ethernet@5 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&enet5>;
+		};
+	};
 };
 
 /include/ "fsl/b4860si-post.dtsi"
diff --git a/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi b/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
index 86032a6..39e4e13 100644
--- a/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
@@ -126,6 +126,41 @@
 
 /include/ "qoriq-qman1.dtsi"
 /include/ "qoriq-bman1.dtsi"
+
+/include/ "qoriq-fman3-0.dtsi"
+/include/ "qoriq-fman3-0-1g-0.dtsi"
+/include/ "qoriq-fman3-0-1g-1.dtsi"
+/include/ "qoriq-fman3-0-1g-2.dtsi"
+/include/ "qoriq-fman3-0-1g-3.dtsi"
+/include/ "qoriq-fman3-0-1g-4.dtsi"
+/include/ "qoriq-fman3-0-1g-5.dtsi"
+	fman0: fman@400000 {
+		/* tx - 1g - 0 */
+		port@a8000 {
+			fsl,qman-channel-id = <0x802>;
+		};
+		/* tx - 1g - 1 */
+		port@a9000 {
+			fsl,qman-channel-id = <0x803>;
+		};
+		/* tx - 1g - 2 */
+		port@aa000 {
+			fsl,qman-channel-id = <0x804>;
+		};
+		/* tx - 1g - 3 */
+		port@ab000 {
+			fsl,qman-channel-id = <0x805>;
+		};
+		/* tx - 1g - 4 */
+		port@ac000 {
+			fsl,qman-channel-id = <0x806>;
+		};
+		/* tx - 1g - 5 */
+		port@ad000 {
+			fsl,qman-channel-id = <0x807>;
+		};
+	};
+
 /include/ "qoriq-mpic.dtsi"
 
 	guts: global-utilities@e0000 {
diff --git a/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi b/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi
index 33bc600..04dce81 100644
--- a/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi
@@ -50,6 +50,18 @@
 		dma0 = &dma0;
 		dma1 = &dma1;
 		sdhc = &sdhc;
+
+		fman0 = &fman0;
+		crypto = &crypto;
+		sec_jr0 = &sec_jr0;
+		sec_jr1 = &sec_jr1;
+		sec_jr2 = &sec_jr2;
+		sec_jr3 = &sec_jr3;
+		rtic_a = &rtic_a;
+		rtic_b = &rtic_b;
+		rtic_c = &rtic_c;
+		rtic_d = &rtic_d;
+		sec_mon = &sec_mon;
 	};
 
 	cpus {
-- 
1.7.9.7

